.pylintrc

```diff
@@ -67,7 +67,7 @@ ignore-imports=yes
 
 [MESSAGES CONTROL]
 
-disable=consider-using-ternary,locally-disabled,locally-enabled,logging-not-lazy,abstract-method,arguments-differ,broad-except,duplicate-code,fixme,inconsistent-return-statements,len-as-condition,missing-raises-doc,multiple-constructor-doc,no-else-return,no-member,no-self-use,not-context-manager,redefined-variable-type,redundant-returns-doc,too-many-arguments,too-many-boolean-expressions,too-many-branches,too-many-instance-attributes,too-many-lines,too-many-locals,too-many-public-methods,too-many-nested-blocks,too-many-statements,wrong-import-order
+disable=consider-using-ternary,locally-disabled,locally-enabled,logging-not-lazy,abstract-method,arguments-differ,broad-except,duplicate-code,fixme,inconsistent-return-statements,len-as-condition,missing-raises-doc,multiple-constructor-doc,no-else-return,no-member,no-self-use,not-context-manager,redefined-variable-type,redundant-returns-doc,too-many-arguments,too-many-boolean-expressions,too-many-branches,too-many-instance-attributes,too-many-lines,too-many-locals,too-many-public-methods,too-many-nested-blocks,too-many-statements,wrong-import-order,import-only-modules
```

Comment:
I think we do want to enforce import-only-modules. Why remove it?

In general changes should not lead to additional checks being disabled.

Reply:
Actually the imports from the future modules are not modules(for example from __future__ import absolute_import), that's why I disabled the check

Reply:
Then you could call `import __future__ as future` and use `future.absolute_import` in code.

Reply:
I think these imports work on the codebase automatically (i.e we don't need to write future.print_function to print a statement, a simple print would suffice) Hence we cannot import at module level. The other option is to disable pylint for these statements. What do you think?

Reply:
Sorry, I don't understand your comment. I am suggesting importing at the module level and referring to functions using the module name as a prefix, so that we do not have to modify .pylintrc. So I am not sure I understand what you mean by "hence we cannot import at module level".

If the concern is stuff like `print` then you could put that in python_utils and have other modules call python_utils.print(). But I'm not sure I understand what exactly the issue is here...

Reply:
I mean that absolute_import, print_function, etc are not modules. The import statement here is more of a "Future" statement -- they change how the Python module is parsed, which is why they must be at the top of the file. They give new or different meaning to words or symbols in the file. This is why we can not use the functions from the __future__ module if its imported at module level. If you know of any way to do it, please tell me. Thanks

Reply:
Ah, sorry, I see what you mean. OK, in that case, let's keep the lint config as-is and use a #pylint-disable in the relevant file. That would still preserve the requirement that all non-future imports be at the module level. Would that be OK?

Reply:
Yes sounds great. Done